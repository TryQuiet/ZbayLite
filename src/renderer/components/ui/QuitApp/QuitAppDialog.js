import React from 'react'
import PropTypes from 'prop-types'
import * as R from 'ramda'

import Dialog from '@material-ui/core/Dialog'
import DialogContent from '@material-ui/core/DialogContent'
import DialogActions from '@material-ui/core/DialogActions'
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'
import { withStyles } from '@material-ui/core/styles'
const styles = theme => ({
  root: {},
  info: {
    fontWeight: 500
  },
  dialogContent: {
    borderBottom: `1px solid ${theme.palette.colors.veryLightGray}`
  },
  buttonNo: {
    borderRight: `1px solid ${theme.palette.colors.veryLightGray}`,
    cursor: 'pointer',
    padding: 10
  },
  buttonYes: {
    cursor: 'pointer',
    padding: 10,
    color: theme.palette.colors.lushSky
  },
  dialogActions: {
    padding: 0
  },
  typography: {
    textAlign: 'center'
  }
})
export const QuitAppDialog = ({ classes, handleClose, handleQuit, open }) => (
  <Dialog onClose={handleClose} open={open}>
    <DialogContent className={classes.dialogContent}>
      <Typography className={classes.info} variant='body2'>
        Do you want to quit Zbay?
      </Typography>
    </DialogContent>
    <DialogActions className={classes.dialogActions}>
      <Grid container direction='row' justify='flex-end'>
        <Grid item xs className={classes.buttonNo} onClick={handleClose}>
          <Typography className={classes.typography} variant='body1'>
            No
          </Typography>
        </Grid>
        <Grid item xs className={classes.buttonYes} onClick={handleQuit}>
          <Typography className={classes.typography} variant='body1'>
            Yes
          </Typography>
        </Grid>
      </Grid>
    </DialogActions>
  </Dialog>
)

QuitAppDialog.propTypes = {
  handleClose: PropTypes.func.isRequired,
  handleQuit: PropTypes.func.isRequired,
  open: PropTypes.bool.isRequired
}

export default R.compose(
  React.memo,
  withStyles(styles)
)(QuitAppDialog)
