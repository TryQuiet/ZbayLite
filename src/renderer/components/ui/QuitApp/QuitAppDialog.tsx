import React from 'react'

import Dialog from '@material-ui/core/Dialog'
import DialogContent from '@material-ui/core/DialogContent'
import DialogActions from '@material-ui/core/DialogActions'
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'
import { makeStyles } from '@material-ui/core/styles'

const useStyles = makeStyles(theme => ({
  root: {},
  info: {
    fontWeight: 500
  },
  dialogContent: {
    borderBottom: `1px solid ${theme.palette.colors.veryLightGray}`
  },
  buttonNo: {
    borderRight: `1px solid ${theme.palette.colors.veryLightGray}`,
    cursor: 'pointer',
    padding: 10
  },
  buttonYes: {
    cursor: 'pointer',
    padding: 10,
    color: theme.palette.colors.lushSky
  },
  dialogActions: {
    padding: 0
  },
  typography: {
    textAlign: 'center'
  }
}))

interface QuitAppDialogProps {
  handleClose: () => void
  handleQuit: () => void
  open: boolean
}

export const QuitAppDialog: React.FC<QuitAppDialogProps> = ({ handleClose, handleQuit, open }) => {
  const classes = useStyles({})
  return (
    <Dialog onClose={handleClose} open={open}>
      <DialogContent className={classes.dialogContent}>
        <Typography className={classes.info} variant='body2'>
          Do you want to quit Zbay?
        </Typography>
      </DialogContent>
      <DialogActions className={classes.dialogActions}>
        <Grid container direction='row' justify='flex-end'>
          <Grid item xs className={classes.buttonNo} onClick={handleClose}>
            <Typography className={classes.typography} variant='body1'>
              No
            </Typography>
          </Grid>
          <Grid item xs className={classes.buttonYes} onClick={handleQuit}>
            <Typography className={classes.typography} variant='body1'>
              Yes
            </Typography>
          </Grid>
        </Grid>
      </DialogActions>
    </Dialog>
  )
}

export default QuitAppDialog
