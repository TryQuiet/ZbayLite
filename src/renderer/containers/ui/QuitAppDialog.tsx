import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import * as R from 'ramda'

import { withModal } from '../../store/handlers/modals'
import QuitAppDialog from '../../components/ui/QuitAppDialog'
import { remote } from 'electron'

export const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      handleQuit: () => () => {
        remote.app.relaunch()
        remote.app.quit()
      }
    },
    dispatch
  )

export default R.compose(
  connect(null, mapDispatchToProps),
  withModal('quitApp')
)(QuitAppDialog)
