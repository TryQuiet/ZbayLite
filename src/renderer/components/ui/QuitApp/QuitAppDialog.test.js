/* eslint import/first: 0 */
import React from 'react'
import { shallow } from 'enzyme'

import { QuitAppDialog } from './QuitAppDialog'
import { mockClasses } from '../../../../shared/testing/mocks'

describe('QuitAppDialog', () => {
  it('renders component', () => {
    const result = shallow(
      <QuitAppDialog
        open
        classes={mockClasses}
        handleClose={jest.fn()}
        handleQuit={jest.fn()}
      />
    )
    expect(result).toMatchSnapshot()
  })
})
