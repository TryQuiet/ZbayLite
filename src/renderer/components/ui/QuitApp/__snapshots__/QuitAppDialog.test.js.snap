// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuitAppDialog renders component 1`] = `
<WithStyles(ForwardRef(Dialog))
  onClose={[MockFunction]}
  open={true}
>
  <WithStyles(ForwardRef(DialogContent))
    className="dialogContent-mock"
  >
    <WithStyles(ForwardRef(Typography))
      className="info-mock"
      variant="body2"
    >
      Do you want to quit Zbay?
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(DialogContent))>
  <WithStyles(ForwardRef(DialogActions))
    className="dialogActions-mock"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      direction="row"
      justify="flex-end"
    >
      <WithStyles(ForwardRef(Grid))
        className="buttonNo-mock"
        item={true}
        onClick={[MockFunction]}
        xs={true}
      >
        <WithStyles(ForwardRef(Typography))
          className="typography-mock"
          variant="body1"
        >
          No
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        className="buttonYes-mock"
        item={true}
        onClick={[MockFunction]}
        xs={true}
      >
        <WithStyles(ForwardRef(Typography))
          className="typography-mock"
          variant="body1"
        >
          Yes
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(DialogActions))>
</WithStyles(ForwardRef(Dialog))>
`;
